src/regex/pcre2.cr

```diff
@@ -0,0 +1,176 @@
+require "./lib_pcre2"
+
+# :nodoc:
+module Regex::PCRE2
+  @re : LibPCRE2::Code*
+
+  # :nodoc:
+  def initialize(*, _source @source : String, _options @options)
+    @re = PCRE2.compile(source, pcre2_options(options) | LibPCRE2::UTF | LibPCRE2::NO_UTF_CHECK | LibPCRE2::DUPNAMES | LibPCRE2::UCP) do |error_message|
+      raise ArgumentError.new(error_message)
+    end
+  end
+
+  protected def self.compile(source, options)
+    if res = LibPCRE2.compile(source, source.bytesize, options, out errorcode, out erroroffset, nil)
+      res
+    else
+      message = String.new(256) do |buffer|
+        bytesize = LibPCRE2.get_error_message(errorcode, buffer, 256)
+        {bytesize, 0}
+      end
+      yield "#{message} at #{erroroffset}"
+    end
+  end
+
+  private def pcre2_options(options)
+    flag = 0
+    options.each do |option|
+      flag |= case option
+              when .ignore_case?   then LibPCRE2::CASELESS
+              when .multiline?     then LibPCRE2::DOTALL | LibPCRE2::MULTILINE
+              when .extended?      then LibPCRE2::EXTENDED
+              when .anchored?      then LibPCRE2::ANCHORED
+              when .utf_8?         then LibPCRE2::UTF
+              when .no_utf8_check? then LibPCRE2::NO_UTF_CHECK
+              when .dupnames?      then LibPCRE2::DUPNAMES
+              when .ucp?           then LibPCRE2::UCP
+              else
+                raise "unreachable"
```

Comment:
Any reason to not use an exhaustive case?
```suggestion
              in .ignore_case?   then LibPCRE2::CASELESS
              in .multiline?     then LibPCRE2::DOTALL | LibPCRE2::MULTILINE
              in .extended?      then LibPCRE2::EXTENDED
              in .anchored?      then LibPCRE2::ANCHORED
              in .utf_8?         then LibPCRE2::UTF
              in .no_utf8_check? then LibPCRE2::NO_UTF_CHECK
              in .dupnames?      then LibPCRE2::DUPNAMES
              in .ucp?           then LibPCRE2::UCP
```

Reply:
I think this is a flags enum

Reply:
Yes, it's a flags enum. So no exhaustiveness.